20 admin main view

EliasUR opened the pull request:
# Description
What did you implemented/Why did you implemented this:
What?:
- Added Stylized view to employees CRUD and search method to filter employees by name

Why?:
- The styles and search input were added to make easier the navigation in Employees view

  ## Screenshots
![image](https://user-images.githubusercontent.com/59623052/158913996-e9f175f6-3931-4b57-9248-32a01bb920ea.png)
![image](https://user-images.githubusercontent.com/59623052/158914117-2c1f2fdb-e60f-4106-9987-10967d4c5d6b.png)

EliasUR committed:
Add sass variables to project

EliasUR committed:
Add first design to employees index

EliasUR committed:
Add first styles to new_ employee view

EliasUR committed:
Refactor employees_controller methods using the new database

EliasUR committed:
Add a partial form to employees index

Was added a partial form to employees/index which makes both edit and
create actions

EliasUR committed:
Add final design to employees_form

Added final design to employees' form and corrected indentation in some
files

EliasUR committed:
Complete Employees View

Changed name of show method which made the deactivate fuction to  update_status method. Added a new route for update_status.

AlejandroCuriel commented on app/assets/stylesheets/employees/form.scss:
```diff
@@ -0,0 +1,14 @@
+.employee_form{
+  width: 100%;
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: space-between;
+  align-items: space-around;
```
it is good development practice to alphabetize styles

EliasUR replied:
Ok. I just have a doubt, "align items" is a flex property. Should I put this before or after "display:flex"?.

EliasUR replied:
change added

AlejandroCuriel commented on app/assets/stylesheets/employees/form.scss:
```diff
@@ -0,0 +1,14 @@
+  align-items: space-around;
+
+  .form-group{
+    width: 30%;
+    display: flex;
+    flex-direction: column;
+    justify-content: space-around;
```
Order this block

EliasUR replied:
change added

AlejandroCuriel commented on app/assets/stylesheets/employees/button.scss:
```diff
@@ -0,0 +1,9 @@
+.btn{
+  min-width: 72px;
+}
+.input-group-append{
```
if it's a new tag, put a space

EliasUR replied:
change added

AlejandroCuriel reviewed (changes requested):
correct those small details for a better clean code

vyk2rr commented on app/controllers/employees_controller.rb:
```diff
@@ -1,34 +1,64 @@
   def create
-    Employee.create name: params['name'], position: params['position'], email: ['email'],
-                    private_number: ['private_number']
+    name = params[:employee][:name].upcase
+    Employee.create name: name, role_id: params['role'], 
+                    email: params[:employee][:email], branch_id: params['branch'], 
+                    private_number: params[:employee][:private_number], status: true
```
I think all the logic in this create and update methods is a little bit hard to understand, probably there are things that needs to live in the model instead of the controller. Quick example => https://riptutorial.com/ruby-on-rails/example/9609/fat-model--skinny-controller

And as well, I think it's more clear and easy to read the code when the functions actually return a value, instead of assigning @variables. The method new_form introduces difficulty to read the controller 

And last but not least important, it doesn't matter if you leave it like that but at least it would be of a lot of help to understand in the future the output of this controller if you add a unitary test with rspec, and in that way if someone in the future modifies this controller or any of your models, the rspec will ensure that everything remains working as normal.

EliasUR replied:
Thanks, I'm going to make some changes

EliasUR replied:
I've added this changes to EmployeesController and i'm going to add some tests later. Thank you again.

EliasUR committed:
Made some corrections

scss files was ordered alphabetically and change complexity in some
EmployeesController's actionss using Fat Model, Skinny Controller.

juperameza reviewed (dismissed):
LGTM :) ♥

AlejandroCuriel reviewed (dismissed):
LGTM :robot: :rocket:

femm34 reviewed (dismissed):
LGTM 😀

JuanequeX commented on app/assets/stylesheets/employees/box.scss:
```diff
@@ -0,0 +1,4 @@
+.box{
```
Add an extra space between your class and your bracket

EliasUR replied:
Added

JuanequeX commented on app/assets/stylesheets/employees/button.scss:
```diff
@@ -0,0 +1,10 @@
+.btn{
```
Add an extra space between your class and your bracket

JuanequeX commented on app/assets/stylesheets/employees/button.scss:
```diff
@@ -0,0 +1,10 @@
+  min-width: 72px;
+}
+
+.input-group-append{
```
Add an extra space between your class and your bracket

EliasUR replied:
Added

JuanequeX commented on app/assets/stylesheets/employees/button.scss:
```diff
@@ -0,0 +1,10 @@
+}
+
+.input-group-append{
+  .btn{
```
Add an extra space between your class and your bracket

JuanequeX commented on app/assets/stylesheets/employees/form.scss:
```diff
@@ -0,0 +1,14 @@
+.employee_form{
```
Add an extra space between your class and your bracket

EliasUR replied:
Added

JuanequeX commented on app/assets/stylesheets/employees/form.scss:
```diff
@@ -0,0 +1,14 @@
+  justify-content: space-between;
+  width: 100%;
+
+  .form-group{
```
Add an extra space between your class and your bracket

JuanequeX commented on app/assets/stylesheets/variables.scss:
```diff
@@ -0,0 +1,5 @@
+$light-blue: #03A9F4;
+$main-blue: #677FFF;
+$rounded: 25px;
+$shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
```
Add a blank space here please

EliasUR replied:
File deleted because we don't use it anymore

JuanequeX commented:
you have conflicts with your branch to merge them, solve them in your local code, they can use the terminal with the help of this [resource](https://docs.openstack.org/doc-contrib-guide/additional-git-workflow/rebase.html)

EliasUR committed:
Add change from master

EliasUR committed:
Add some corretions in stylesheets files

VictorElizalde reviewed (approved):
LGTM

femm34 reviewed (approved):
LGTM :smile:

JuanequeX reviewed (approved):
LGTM
